src/ml_project/train.py

```diff
@@ -16,63 +16,33 @@
 class TrainModel():
     def __init__(self, cfg: DictConfig) -> None:
         self.cfg = cfg
-        self.run_dir = cfg.run_dir
+        processed_dir = "data/processed"
+        self.processed_data_path = osp.join(processed_dir, cfg.data.dataset_name)
+
         self._set_seed(cfg.train.random_seed)  # Set seed for reproducibility
-        self.processed_data_path = osp.join(cfg.data.processed_dir, cfg.data.dataset_name)
         self._check_data_path()
-        self._make_run_dir_folders()
         self._make_logger()
         self._save_data()
-      
-    def _make_run_dir_folders(self) -> None:
-        # Check if run directory exists
-        if not osp.exists(self.run_dir):
-            raise FileNotFoundError(f"Run directory {self.run_dir} does not exist. Please create it first.")
-
-        # Store model
-        self.run_dir_model = osp.join(self.run_dir, "models")
-        os.makedirs(self.run_dir_model, exist_ok=False)
-        
-        # Store figures
-        self.run_dir_figures = osp.join(self.run_dir, "figures")
-        os.makedirs(self.run_dir_figures, exist_ok=False)
-        
-        # Store wandb logs
-        self.run_dir_wandb = osp.join(self.run_dir, "wandb")
-        os.makedirs(self.run_dir_wandb, exist_ok=False)
-
-        # Store dataset
-        self.run_dir_data = osp.join(self.run_dir, "data")
-        os.makedirs(self.run_dir_data, exist_ok=False)
 
+    def _make_logger(self) -> None:
+        """Create a logger object."""
+        # Initialize wandb and set log directory
+        self.wandb_run = wandb.init(
+            project=self.cfg.wandb.project_name,
+            config=OmegaConf.to_container(self.cfg, resolve=True))
+    
     def _save_data(self) -> None:
```

Comment:
```suggestion
    def _save_wandb_data(self) -> None:
```

In situations like this I think it's nice to be more explicit